// @ionic-native/pro/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { IonicNativePlugin, cordova, checkAvailability, cordovaInstance } from '@ionic-native/core';
import { Observable } from 'rxjs';
var ProDeploy = /** @class */ (function () {
    function ProDeploy(_objectInstance) {
        this._objectInstance = _objectInstance;
    }
    ProDeploy.prototype.init = function (config) { return cordovaInstance(this, "init", {}, arguments); };
    ProDeploy.prototype.check = function () { return cordovaInstance(this, "check", {}, arguments); };
    ProDeploy.prototype.download = function () { return cordovaInstance(this, "download", { "observable": true }, arguments); };
    ProDeploy.prototype.extract = function () { return cordovaInstance(this, "extract", { "observable": true }, arguments); };
    ProDeploy.prototype.redirect = function () { return cordovaInstance(this, "redirect", {}, arguments); };
    ProDeploy.prototype.info = function () { return cordovaInstance(this, "info", {}, arguments); };
    ProDeploy.prototype.getVersions = function () { return cordovaInstance(this, "getVersions", {}, arguments); };
    ProDeploy.prototype.deleteVersion = function (version) { return cordovaInstance(this, "deleteVersion", {}, arguments); };
    return ProDeploy;
}());
export { ProDeploy };
var ProOriginal = /** @class */ (function (_super) {
    __extends(ProOriginal, _super);
    function ProOriginal() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ProOriginal.prototype.deploy = function () {
        var _this = this;
        return (function () {
            if (checkAvailability(_this) === true) {
                if (_this._deploy) {
                    return _this._deploy;
                }
                else {
                    _this._deploy = new ProDeploy(ProOriginal.getPlugin().deploy);
                    return _this._deploy;
                }
            }
        })();
    };
    ProOriginal.prototype.enableCrashLogging = function () { return cordova(this, "enableCrashLogging", {}, arguments); };
    ProOriginal.prototype.checkForPendingCrash = function () { return cordova(this, "checkForPendingCrash", {}, arguments); };
    ProOriginal.prototype.loadPendingCrash = function () { return cordova(this, "loadPendingCrash", {}, arguments); };
    ProOriginal.prototype.forceCrash = function () { return cordova(this, "forceCrash", {}, arguments); };
    ProOriginal.prototype.getAppInfo = function () { return cordova(this, "getAppInfo", {}, arguments); };
    ProOriginal.pluginName = "Pro";
    ProOriginal.plugin = "cordova-plugin-ionic";
    ProOriginal.pluginRef = "IonicCordova";
    ProOriginal.repo = "https://github.com/ionic-team/cordova-plugin-ionic";
    ProOriginal.platforms = ["Android", "iOS"];
    ProOriginal.install = "ionic cordova plugin add cordova-plugin-ionic --save --variable APP_ID=\"XXXXXXXX\" --variable CHANNEL_NAME=\"Channel\"";
    return ProOriginal;
}(IonicNativePlugin));
var Pro = new ProOriginal();
export { Pro };
