// @ionic-native/pro/ngx/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova, checkAvailability, cordovaInstance } from '@ionic-native/core';
import { Observable } from 'rxjs';
var ProDeploy = /** @class */ (function () {
    function ProDeploy(_objectInstance) {
        this._objectInstance = _objectInstance;
    }
    ProDeploy.prototype.init = function (config) { return cordovaInstance(this, "init", {}, arguments); };
    ProDeploy.prototype.check = function () { return cordovaInstance(this, "check", {}, arguments); };
    ProDeploy.prototype.download = function () { return cordovaInstance(this, "download", { "observable": true }, arguments); };
    ProDeploy.prototype.extract = function () { return cordovaInstance(this, "extract", { "observable": true }, arguments); };
    ProDeploy.prototype.redirect = function () { return cordovaInstance(this, "redirect", {}, arguments); };
    ProDeploy.prototype.info = function () { return cordovaInstance(this, "info", {}, arguments); };
    ProDeploy.prototype.getVersions = function () { return cordovaInstance(this, "getVersions", {}, arguments); };
    ProDeploy.prototype.deleteVersion = function (version) { return cordovaInstance(this, "deleteVersion", {}, arguments); };
    return ProDeploy;
}());
export { ProDeploy };
var Pro = /** @class */ (function (_super) {
    __extends(Pro, _super);
    function Pro() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Pro.prototype.deploy = function () {
        var _this = this;
        return (function () {
            if (checkAvailability(_this) === true) {
                if (_this._deploy) {
                    return _this._deploy;
                }
                else {
                    _this._deploy = new ProDeploy(Pro.getPlugin().deploy);
                    return _this._deploy;
                }
            }
        })();
    };
    Pro.prototype.enableCrashLogging = function () { return cordova(this, "enableCrashLogging", {}, arguments); };
    Pro.prototype.checkForPendingCrash = function () { return cordova(this, "checkForPendingCrash", {}, arguments); };
    Pro.prototype.loadPendingCrash = function () { return cordova(this, "loadPendingCrash", {}, arguments); };
    Pro.prototype.forceCrash = function () { return cordova(this, "forceCrash", {}, arguments); };
    Pro.prototype.getAppInfo = function () { return cordova(this, "getAppInfo", {}, arguments); };
    Pro.pluginName = "Pro";
    Pro.plugin = "cordova-plugin-ionic";
    Pro.pluginRef = "IonicCordova";
    Pro.repo = "https://github.com/ionic-team/cordova-plugin-ionic";
    Pro.platforms = ["Android", "iOS"];
    Pro.install = "ionic cordova plugin add cordova-plugin-ionic --save --variable APP_ID=\"XXXXXXXX\" --variable CHANNEL_NAME=\"Channel\"";
    Pro = __decorate([
        Injectable()
    ], Pro);
    return Pro;
}(IonicNativePlugin));
export { Pro };
